### PR TITLE
improve azure storage account creation

### PR DESCRIPTION
**Pull request by zreigz**
## Summary
Check if existing SA belongs to the resource group

## Checklist

- [ ] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [ ] I have added relevant labels to this PR to help with categorization for release notes.

### CODE REVIEW

File: pkg/provider/azure.go [hunk 1]
```diff
@@ -315,7 +319,27 @@ func (az *AzureProvider) Decommision(node *v1.Node) error {
...
 func (az *AzureProvider) getStorageAccount(account string) (storage.Account, error) {
-	return az.clients.Accounts.GetProperties(context.Background(), az.resourceGroup, account, storage.AccountExpandBlobRestoreStatus)
+	ctx := context.Background()
+	saList, err := az.clients.Accounts.List(ctx)
+	if err != nil {
+		return storage.Account{}, err
+	}
+	// check if storage account belongs to the same resource group
+	for _, sa := range saList.Values() {
+		if *sa.Name == account {
```
Comment (michaeljguarino) [inline pkg/provider/azure.go:329]: Why does this work?

Comment (zreigz) [reply pkg/provider/azure.go:329]: Dereferencing a pointer gives us access to the value the pointer points to

Comment (michaeljguarino) [reply pkg/provider/azure.go:329]: oh no not that part, ike how does the logic itself work?

Comment (michaeljguarino) [reply pkg/provider/azure.go:329]: You're listing storage accounts in the subscription then somehow getting the resource group by parsing the full path name or somnething?  Is there no reference to it elsewhere?

Comment (michaeljguarino) [reply pkg/provider/azure.go:329]: Wow, that's nuts. Azure really is trash then

**Review by michaeljguarino — APPROVED** [review]

**Pull request merged**